Jetcaster/app/src/main/java/com/example/jetcaster/ui/podcast/PodcastDetailsScreen.kt

```diff
@@ -0,0 +1,336 @@
+        Row(
+            verticalAlignment = Alignment.Bottom,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            AsyncImage(
+                model = ImageRequest.Builder(LocalContext.current)
+                    .data(podcast.imageUrl)
+                    .crossfade(true)
+                    .build(),
+                contentDescription = stringResource(id = R.string.cd_podcast_image),
+                contentScale = ContentScale.Crop,
+                modifier = Modifier
+                    .size(148.dp)
+                    .clip(MaterialTheme.shapes.large)
+            )
+            Column(
+                modifier = Modifier.padding(start = 16.dp)
+            ) {
+                Text(
+                    text = podcast.title,
+                    maxLines = 2,
+                    overflow = TextOverflow.Ellipsis,
+                    style = MaterialTheme.typography.headlineMedium
+                )
+                PodcastDetailsHeaderItemButtons(
+                    isSubscribed = podcast.isSubscribed ?: false,
+                    onClick = {
+                        toggleSubscribe(podcast)
+                    },
+                    modifier = Modifier.fillMaxWidth()
+                )
+            }
+        }
+        PodcastDetailsDescription(
+            podcast = podcast,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 16.dp)
+        )
+    }
+}
+
+@Composable
+fun PodcastDetailsDescription(
+    podcast: PodcastInfo,
+    modifier: Modifier
+) {
+    var isExpanded by remember { mutableStateOf(false) }
+    var showSeeMore by remember { mutableStateOf(false) }
+    Box(modifier = modifier) {
+        Text(
+            text = podcast.description,
+            style = MaterialTheme.typography.bodyMedium,
+            maxLines = if (isExpanded) Int.MAX_VALUE else 3,
+            overflow = TextOverflow.Ellipsis,
+            onTextLayout = { result ->
+                showSeeMore = result.hasVisualOverflow
+            },
+        )
+        if (showSeeMore) {
+            Box(
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .background(MaterialTheme.colorScheme.surface)
+            ) {
+                // TODO: Add gradient effect
+                Text(
+                    text = stringResource(id = R.string.see_more),
+                    style = MaterialTheme.typography.bodyMedium,
+                    modifier = Modifier
+                        .padding(start = 16.dp)
+                        .clickable {
+                            isExpanded = !isExpanded
+                        }
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun PodcastDetailsHeaderItemButtons(
+    isSubscribed: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(modifier.padding(top = 16.dp)) {
+        Button(
+            onClick = onClick
```

Comment:
You should double-check this but I think this is going to result in TalkBack repeating itself. Since the icon and the text convey the same info, it's probably better to use null for the content description of the icon and maybe merge the semantics at the button.

Reply:
Makes sense to set the icon to `null` and have the Button merge semantics.